src/analyses/goto_check.cpp

```diff
@@ -1342,8 +1342,7 @@ void goto_checkt::check_rec(const exprt &expr, guardt &guard, bool address)
         guard.add(op);
     }
 
-    guard.swap(old_guard);
-
+    guard = std::move(old_guard);
```

Comment:
Almost randomly poking at commits: what's the improvement here? (I actually have a yet-to-be-PR'ed commit lying around making the opposite change, i.e., replacing uses of `std::move` by `swap`.)

Reply:
The new implementation of guardt doesn't have a `swap` (though exprt has). And since `old_guard` is not used afterward, I don't see a reason for using swap?

Reply:
For `irept`-based implementations `= std::move(...)` is at best as efficient as `swap`, and possibly less efficient if `USE_MOVE` is not enabled. (But I do concede that we should just get rid of `USE_MOVE` as an optional feature as we require C++11 anyway.) So in that setting the change doesn't seem to provide any added value.

With the new implementation of `guardt` not being an `irept`, the situation of course is different. But that needs to be spelled out in the commit message...